data_retriever.rb

```diff
@@ -0,0 +1,130 @@
+require 'nokogiri'
+require 'rubygems'
+require 'open-uri'
+require 'watir'
+require 'pry'
+require 'json'
+require_relative "Accounts.rb"
+require_relative "Transactions.rb"
+
+Selenium::WebDriver::Chrome.driver_path="/usr/local/bin/chromedriver"
+
+class Data_Retriever
+
+  def initialize()
+  #attr_reader :available_accounts, :transactions_records
+  @available_accounts = []
+  @transactions_records = []
+  @browser = Watir::Browser.new :chrome
+  end
+  #Opens the browser
+
+  def browse(url)
+    @browser.goto(url)
+    while @browser.div(:id => "oCAPICOM").exists? do sleep 1 end
+  end
+
+  #Does the authentification
+
+  def login
+    @browser.a(class: 'ng-binding').click
+    @browser.a(id: 'demo-link').click
+    @browser.element(:id => "step1").wait_until(&:present?)
+  end
+
+  #Parses
+
+  def parsing
+    #@page = Nokogiri::HTML.parse(@browser.html)
+    @page = Nokogiri.HTML(@browser.html)
+    puts "Parsing is done."
+  end
+
+  #Gathers credit card information
+
+  def gather_credit_info
+    credit_accounts = @page.css("table[id='dashboardAccounts'] tr[id]")
+    @ca_index = 1
+    while @ca_index <= credit_accounts.count do
+      #account_name = @page.css("#dashboardAccounts td.ng-scope.icon-two-line-col > div > div > a > span").text
+      account_name = @page.search("#dashStep1 tr:nth-of-type(#{@ca_index}) td:nth-of-type(1) span").text
+      account_available_balance = @page.css("#dashStep1 tr:nth-of-type(#{@ca_index}) td:nth-of-type(3) span").text
+      account_currency = @page.css("#dashStep1 tr:nth-of-type(#{@ca_index}) td:nth-of-type(2) span").text
+      ca = Accounts.new(account_name, account_available_balance, account_currency,"Credit")
+      @available_accounts << ca
+      @ca_index += 1
+    end
+  end
+
+  #Gathers debit card information
+
+  def gather_debit_info
+    debit_accounts = @page.css("#dashStep2 tr[id]")
+    @da_index = 1
+    while @da_index <= debit_accounts.count do
+      account_name = @page.css("#dashStep2 tr:nth-of-type(#{@da_index}) td:nth-of-type(1) span").text
+      account_currency = @page.css("#dashStep2 tr:nth-of-type(#{@da_index}) td:nth-of-type(3) span").text
+      account_available_balance = @page.css("#dashStep2 tr:nth-of-type(#{@da_index}) td:nth-of-type(2) span").text
+      da = Accounts.new(account_name, account_currency , account_available_balance,"Debit" )

```

Comment:
avoid to using 1, 2 letters variable  
try to use more descriptive variable
left space after all commas

```ruby
debit_account = Accounts.new(account_name, account_currency, account_available_balance, "Debit" )